src/d/actor/d_a_player_sword.inc

```diff
@@ -1106,23 +1285,285 @@ BOOL daPy_lk_c::procCutExB() {
+        }
+        if (frameCtrl.checkPass(daPy_HIO_cutTurn_c0::m.field_0x34)) {
+            m331C.field_0x4 = 0x80;
+            m332C.field_0x4 = 0x80;
+            m333C.field_0x4 = 0x80;
+        } else {
+            if (m331C.field_0x4 == 0x80) {
+                m331C.field_0x4 = 0;
+                m332C.field_0x4 = 0;
+                m333C.field_0x4 = 0;
+            }
+        }
+        m34C2 = 1;
+        return true;
+    }
 }
 
 /* 80158460-801585D8       .text procCutRoll_init__9daPy_lk_cFv */
 BOOL daPy_lk_c::procCutRoll_init() {
-    /* Nonmatching */
+    commonProcInit(daPyProc_CUT_ROLL_e);
+    setSingleMoveAnime(ANM_CUTTURNB, daPy_HIO_cutRoll_c0::m.field_0x8,
+                       daPy_HIO_cutRoll_c0::m.field_0xC, daPy_HIO_cutRoll_c0::m.field_0x0,
+                       daPy_HIO_cutRoll_c0::m.field_0x10);
+    m3578 = 0;
+    seStartMapInfo(JA_SE_LK_V_KAITEN_S);
+    dComIfGp_setItemMagicCount(-2);
+    mCutType = 9;
+    mCutCount = 0;
+    offResetFlg0(daPyRFlg0_UNK8000000);
+    if (checkNormalSwordEquip()) {
+        m35A4 = daPy_HIO_cutRoll_c0::m.field_0x20;
+        if (checkNoResetFlg1(daPyFlg1_UNK8000)) {
+            mAtCyl.SetAtAtp(4);
+        } else {
+            mAtCyl.SetAtAtp(2);
+        }
+    } else {
+        m35A4 = daPy_HIO_cutRoll_c0::m.field_0x24;
+        if (checkNoResetFlg1(daPyFlg1_UNK8000)) {
+            mAtCyl.SetAtAtp(8);
+        } else {
+            mAtCyl.SetAtAtp(4);
+        }
+    }
+    mAtCyl.SetR(m35A4 * 0.5f);
+    m34D0 = daPy_HIO_cutRoll_c0::m.field_0x2;
+    current.angle.y = shape_angle.y;
+    m34BE = 4;
+    dComIfGp_setPlayerStatus0(0, daPyStts0_SPIN_ATTACK_e);
+    endFlameDamageEmitter();
+    return true;
 }
 
 /* 801585D8-801588F0       .text procCutRoll__9daPy_lk_cFv */
 BOOL daPy_lk_c::procCutRoll() {
-    /* Nonmatching */
+    /* Nonmatching - regalloc, math */
+    s16 uVar1;
+    float fVar2;
+    short sVar3;
+    cXyz local_48;
+    cXyz local_3c;
+
+    if (mFrameCtrlUnder[UNDER_MOVE0_e].getRate() < 0.01f) {
+        uVar1 = current.angle.y;
+        if ((mAcch.ChkWallHit()) && (mDemo.getDemoMode() != 0x2b)) {
+            fVar2 = 25.0f + (mAcchCir[0]).GetWallR();
+            local_48.set(current.pos.x, current.pos.y + (mAcchCir[0]).GetWallH(), current.pos.z);
+            local_3c.set(local_48.x + fVar2 * cM_ssin(uVar1), local_48.y,
+                         local_48.z + fVar2 * cM_scos(uVar1));
+            mLinkLinChk.Set(&local_48, &local_3c, this);
+            if (dComIfG_Bgsp()->LineCross(&mLinkLinChk)) {
+                cM3dGPla* triPla = dComIfG_Bgsp()->GetTriPla(mLinkLinChk);
+                shape_angle.y = cM_atan2s(-triPla->GetNP()->x, -triPla->GetNP()->z);
+                current.angle.y = shape_angle.y;
+                procFrontRollCrash_init();
+                goto LAB_801588d8;
+            }
+        }
+        if (mStickDistance > 0.05f) {
+            cLib_addCalcAngleS(&current.angle.y, m34E8, daPy_HIO_move_c0::m.field_0x6,
```

Comment:
I think the goto could be avoided with a huge else block here that covers most of the function
```suggestion
            }
        } else {
            if (mStickDistance > 0.05f) {
                cLib_addCalcAngleS(&current.angle.y, m34E8, daPy_HIO_move_c0::m.field_0x6,
```